### PR TITLE
Support new os GWD

### PR DESCRIPTION
**Pull request by hrtrd**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by hrtrd** [comment]
@PipoCanaja can u approve?

File: includes/polling/os/gwd.inc.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+$hardware = trim(snmp_get($device, '.1.3.6.1.4.1.10072.2.20.1.1.1.1.1.9.1', '-OQv'), '"');
+$version = trim(snmp_get($device, '.1.3.6.1.4.1.10072.2.20.1.1.1.1.1.7.1', '-OQv'), '"');
+$serial = trim(snmp_get($device, '.1.3.6.1.4.1.10072.2.20.1.1.2.1.1.18.1.1', '-OQv'), '"');
+
+if (empty($hardware) && empty($version)) {
+    $hardware =  $hardware;
+    $version = $software;
+    $serial = $serial;
+    unset($temp_data);
+}
```
Comment (PipoCanaja) [inline includes/polling/os/gwd.inc.php]: Hello
Please use snmp_get_multi to poll multiple OIDs (faster).
And remove the 2nd part of the code (if) which does not make sense at all. 
Thx

Comment (hrtrd) [reply includes/polling/os/gwd.inc.php]: if removed, yes. But snmp_get_multi doesnt work for this devices. due to the fact that there is a lot of data behind this id, we get a bunch of unnecessary information on the output. This is alt, and for this id we get data about all onu devices

File: includes/polling/mempools/gwd.inc.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * the source code distribution for details.
+ */
+
+d_echo('gwd: ');
+$perc = snmp_get($device, '1.3.6.1.4.1.10072.2.20.1.1.2.1.1.20.1.1', '-OvQ', 'GW-EPON-MIB');
+if (is_numeric($perc)) {
+    $mempool['perc'] = $perc;
+    $mempool['used'] = $perc;
+    $mempool['total'] = 100;
+    $mempool['free'] = 100 - $perc;
```
Comment (PipoCanaja) [inline includes/polling/mempools/gwd.inc.php]: total, used and free should be "bytes" and not percents. You need to do the conversion depending on the data provided by the device.

File: includes/polling/os/gwd.inc.php (hunk 1 above)
Comment (PipoCanaja) [reply includes/polling/os/gwd.inc.php]: snmp_get_multi != snmpwalk. You get only the OID you ask with snmp_get_multi. Look at other OS polling code to see how it is used.

**Comment by PipoCanaja** [comment]
Could you please also open a new PR to fix "includes/polling/os/cdata.inc.php " that has the same issue in polling code (snmp get multi and wrong "if" code). 
Thanx

**Comment by hrtrd** [comment]
Yeah, sure

File: includes/polling/mempools/gwd.inc.php (hunk 2 above)
Comment (hrtrd) [reply includes/polling/mempools/gwd.inc.php]: done

**Comment by hrtrd** [comment]
hwats wrong now?

File: includes/discovery/mempools/gwd.inc.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+if ($device['os'] == 'gwd') {
+    echo 'GWD-MEMORY-POOL: ';
+
+    $total = snmp_get($device, '.1.3.6.1.4.1.10072.2.20.1.1.2.1.1.28.1.1', '-OvQ', 'GW-EPON-MIB');
+    $usage  = snmp_get($device, '.1.3.6.1.4.1.10072.2.20.1.1.2.1.1.20.1.1', '-OvQ', 'GW-EPON-MIB');
+
```
Comment (PipoCanaja) [inline includes/discovery/mempools/gwd.inc.php]: snmp_get_multi would be better here as well.

File: includes/polling/mempools/gwd.inc.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (c) 2019 hartred <tumanov@asarta.ru>
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+$mempool['total']   = snmp_get($device, ".1.3.6.1.4.1.10072.2.20.1.1.2.1.1.28.1.1", "-Ovq");
+$mempool['used']    = snmp_get($device, ".1.3.6.1.4.1.10072.2.20.1.1.2.1.1.20.1.1", "-Ovq");
```
Comment (PipoCanaja) [inline includes/polling/mempools/gwd.inc.php]: and same here.

**Comment by PipoCanaja** [comment]
And once all code is OK, you'll need to run again the json part (cause the memory pools code was corrected, and you still have the wrong values in the tests right now)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1

**Pull request merged**